### PR TITLE
chore: add zap scan

### PR DESCRIPTION
**Pull request by drewbo**
## Changes proposed in this pull request:
- Adds ZAP scanning for core/admin/queues
- Uploads the results to S3

## security considerations
This adds an admin test user to the database to scan sites, it is removed after scanning.

### CODE REVIEW

File: scripts/create-test-users.js [hunk 1]
```diff
@@ -4,7 +4,7 @@ const factory = require('../test/api/support/factory');
 const { authenticatedSession } = require('../e2e/auth-session');
 
 async function createUsers() {
-  const user = await factory.user({ username: 'test-e2e-user' });
+  const user = await factory.user({ username: process.env.PAGES_TEST_USER || 'generic-test-user' });
```
Comment (apburnes) [inline scripts/create-test-users.js:7]: Do we want to update this user to belong to an org? Manage an org?

File: scripts/count-yaml.sh
Comment (apburnes) [inline scripts/count-yaml.sh:1]: What is the purpose of the count-yaml.sh script?

File: ci/pipeline.yml [hunk 2]
```diff
@@ -516,6 +592,8 @@ resources:
   - name: redis
   - name: postgres
   - name: node
+  - name: zap
+  - name: dind
```
Comment (apburnes) [inline ci/pipeline.yml:596]: Removing the version will make updates across all of our pipelines much easier.

**Review by apburnes — COMMENTED** [review]
Just a few questions but this is a great addition to our process. Many hours per month will be saved for better use. 🏆

File: scripts/count-yaml.sh
Comment (drewbo) [reply scripts/count-yaml.sh:1]: a little fun on the side, will remove

File: scripts/create-test-users.js (hunk 1 above)
Comment (drewbo) [reply scripts/create-test-users.js:7]: we don't really have standardized fixture data for them to test against but having them in an org could be okay

File: scripts/count-yaml.sh
Comment (apburnes) [reply scripts/count-yaml.sh:1]: Oh. Hahah. Keep it.

**Comment by drewbo** [comment]
Issues running this on admin/queues SPAs, closing for now

**Pull request closed without merging**